core/soa/src/main/java/org/opennms/core/soa/lookup/ServiceLookupBuilder.java

```diff
@@ -39,7 +39,7 @@
 
 public static final long GRACE_PERIOD_MS = SystemProperties.getLong("org.opennms.core.soa.lookup.gracePeriodMs", TimeUnit.MINUTES.toMillis(5));
 
-    public static final long WAIT_PERIOD_MS = SystemProperties.getLong("org.opennms.core.soa.lookup.gracePeriodMs", TimeUnit.MINUTES.toMillis(1));
+    public static final long WAIT_PERIOD_MS = SystemProperties.getLong("org.opennms.core.soa.lookup.waitPeriodMs", TimeUnit.MINUTES.toMillis(1));
```

Comment:
Silly question, since I have no idea if anyone overrides `gracePeriodMs` in the real world, but couldn't this cause unexpected behavior since it's getting dropped from 5 to 1 minute?

Should this be changed to fall back to the value of gracePeriodMs if it's set, and only _then_ to 1 minute if both aren't? Or do we care?

Reply:
**Great** catch, and you're right about "couldn't this cause unexpected behavior since it's getting dropped from 5 to 1 minute?". Looking at, I see that there are *two* interrelated issues here with the change to the definition of `WAIT_PERIOD_MS` and using it in `ServiceLookupBuilder.blocking`:
1. We are changing the default wait time of 5 minutes to 1 minute--but only for `ServiceLookupBuilder.blocking` calls, which used `GRACE_PERIOD_MS` before. Note that there are other existing users of `WAIT_PERIOD_MS` that are outside of `ServiceLookupBuilder`.
2. We are changing the property to override from `... gracePeriodMs` to `...waitPeriodMs` (which is completely new) for the wait time for all service lookups.

The first one is which worries me most. I'm not that worried about the second one, as it probably impacts very few, if any, people. 

If we wanted to be really safe, we'd want to have two different fallbacks:
1. One for `WAIT_PERIOD_MS` so that it also checks `...gracePeriodMs` before falling back to the default of 1. This will be for the *existing* consumers of `WAIT_PERIOD_MS`. E.g.: ` WAIT_PERIOD_MS = SystemProperties.getLong("org.opennms.core.soa.lookup.waitPeriodMs", SystemProperties.getLong("org.opennms.core.soa.lookup.gracePeriodMs", TimeUnit.MINUTES.toMillis(1)));`
2. Another one just for `ServiceLookupBuilder.blocking` that does the same thing as above, but with a default of 5 if neither property is found.

Or, we could pull the changes to this file the PR and just merge it separately to `develop`?

FWIW, here are the other files that use `WAIT_PERIOD_MS`.
```
features/telemetry/registry/src/main/java/org/opennms/netmgt/telemetry/protocols/registry/impl/TelemetryAdapterRegistryImpl.java:        this(ServiceLookupBuilder.GRACE_PERIOD_MS, ServiceLookupBuilder.WAIT_PERIOD_MS, ServiceLookupBuilder.LOOKUP_DELAY_MS);
features/telemetry/registry/src/main/java/org/opennms/netmgt/telemetry/protocols/registry/impl/TelemetryConnectorRegistryImpl.java:        this(ServiceLookupBuilder.GRACE_PERIOD_MS, ServiceLookupBuilder.WAIT_PERIOD_MS, ServiceLookupBuilder.LOOKUP_DELAY_MS);
features/telemetry/registry/src/main/java/org/opennms/netmgt/telemetry/protocols/registry/impl/TelemetryListenerRegistryImpl.java:        this(ServiceLookupBuilder.GRACE_PERIOD_MS, ServiceLookupBuilder.WAIT_PERIOD_MS, ServiceLookupBuilder.LOOKUP_DELAY_MS);
features/telemetry/registry/src/main/java/org/opennms/netmgt/telemetry/protocols/registry/impl/TelemetryParserRegistryImpl.java:        this(ServiceLookupBuilder.GRACE_PERIOD_MS, ServiceLookupBuilder.WAIT_PERIOD_MS, ServiceLookupBuilder.LOOKUP_DELAY_MS);
features/telemetry/registry/src/main/java/org/opennms/netmgt/telemetry/protocols/registry/impl/TelemetryServiceRegistryImpl.java:        this(staticServiceSupplier, ServiceLookupBuilder.GRACE_PERIOD_MS, ServiceLookupBuilder.WAIT_PERIOD_MS, ServiceLookupBuilder.LOOKUP_DELAY_MS);
```